rolling/base_indexed.py

```diff
@@ -0,0 +1,90 @@
+import abc
+from collections.abc import Iterator
+
+class RollingIndexed(Iterator):
+    """
+    Baseclass for rolling iterators over _indexed_ or sparse data
+    """
+
+    def __init__(self, index, iterable, window_size, window_type="variable", **kwargs):
```

Comment:
Do you think the `window_type` parameter should be removed for this base class? It looks like it may not be needed as it's implicit that the window can be of variable size depending on the values of the index.

Reply:
Yes, but I added the window type to have a common interface with the base RollingObject class.

I don't mind removing it